Frontend/library/src/Config/Config.ts

```
@@ -742,7 +742,13 @@ export class Config {
                 `Cannot set text setting called ${id} - it does not exist in the Config.enumParameters map.`
             );
         } else {
-            this.optionParameters.get(id).selected = settingValue;
+            const optionSetting = this.optionParameters.get(id);
+            const existingOptions = optionSetting.options;
+            if (!existingOptions.includes(settingValue)) {
+                existingOptions.push(settingValue);
+                optionSetting.options = existingOptions;
+            }
+            optionSetting.selected = settingValue;
         }
```

Few questions:

1. Which bit of code actually calls `setOptionSettingValue`?

2. Is it worth having a `setOrAddOptionSettingValue` - my concern is adding if missing is bit unexpected based on the name `setOptionSettingValue`?

Wdyt?

---

It's only used in two places. One for the preferred codec and one for the streamer ids. My logic for adding it in the existing behaviour is that if you're trying to set the selected value of an option selector, it doesn't really make sense for it to fail silently and not do anything.